qiskit_ibm/runtime/runtime_program.py

```diff
@@ -77,48 +78,43 @@ def __init__(
         self._max_execution_time = max_execution_time
         self._version = version
         self._backend_requirements = backend_requirements or {}
-        self._parameters: List[ProgramParameter] = []
-        self._return_values: List[ProgramResult] = []
-        self._interim_results: List[ProgramResult] = []
+        self._parameters = parameters or {}
+        self._return_values = return_values or {}
+        self._interim_results = interim_results or {}
         self._creation_date = creation_date
         self._is_public = is_public
 
-        if parameters:
-            for param in parameters:
-                self._parameters.append(
-                    ProgramParameter(name=param['name'],
-                                     description=param['description'],
-                                     type=param['type'],
-                                     required=param['required']))
-        if return_values is not None:
-            for ret in return_values:
-                self._return_values.append(ProgramResult(name=ret['name'],
-                                                         description=ret['description'],
-                                                         type=ret['type']))
-        if interim_results is not None:
-            for intret in interim_results:
-                self._interim_results.append(ProgramResult(name=intret['name'],
-                                                           description=intret['description'],
-                                                           type=intret['type']))
-
     def __str__(self) -> str:
-        def _format_common(items: List) -> None:
-            """Add name, description, and type to `formatted`."""
-            for item in items:
-                formatted.append(" "*4 + "- " + item.name + ":")
-                formatted.append(" "*6 + "Description: " + item.description)
-                formatted.append(" "*6 + "Type: " + item.type)
-                if hasattr(item, 'required'):
-                    formatted.append(" "*6 + "Required: " + str(item.required))
+        def _format_common(schema: Dict) -> None:
+            """Add title, description and property details to `formatted`."""
+            if "description" in schema:
+                formatted.append(" "*4 + "Description: {}".format(schema["description"]))
```

Comment:
This formatting doesn't look quite right to me. `self._parameters` is set by 
`parameters = json.loads(response["spec"].get('parameters', '{}'))`, so it would have the format of 
```
{
        "iterations": {
          "description": "Number of iterations to run. Each iteration generates a runs a random circuit.",
          "type": "integer",
          "minimum": 0
        }
      },
      "$schema": "https://json-schema.org/draft/2019-09/schema",
      "required": [
        "iterations"
      ]
    }
```
so you wouldn't find the `description` field?